packages/flutter_tools/test/general.shard/resident_runner_test.dart

```diff
@@ -2670,6 +2760,9 @@ void main() {
       return 0;
     });
     await residentRunner.run();
+    // Await a short delay so that we don't try to exit before all the expected
+    // VM service requests have been fired.
+    await Future<void>.delayed(const Duration(seconds: 1));
```

Comment:
a second is a long time for a unit test. Can you get all the responses through by awaiting null?

Reply:
it does not pass with `await null`. I reduced the delay down to 200 milliseconds though.